### ISSUE TITLE
To-do checklist exp/gold/MP gain bug?

### ISSUE THREAD

**Issue by GrayDays** [HabitRPG/habitica#3474]
If I make a to-do with a checklist that has only one item and check off that item before checking off the to-do, or if I have a checklist with multiple items and check off only one before checking off the to-do, I still gain twice the exp/gold/MP that I would for task without a checklist. Also, if I go into completed to-dos and uncheck the task, only the reward amount that I would receive for a to-do without a checklist is subtracted. This has led to me unintentionally farming experience when I accidentally check off a to-do or when I give up and decide to reward myself at least for what I was able to accomplish on something only partially completed. I could circumvent this by editing the task to be the portion I completed and removing the checklist, but if this is unintentional I figured I should share. I would never think of completing only one part of a large to-do as being worth twice as much as completing the whole thing in one chunk.

If it helps, multipliers for further checklist items seem to progress normally, though they're also high - 1x for no checklist or a checklist with no checked items, 2x for one checked item regardless of the size of the list, 3x for two checked, 4x for three, etc. It seems as if the task itself is being counted as an extra checklist item.

**Comment by deilann** [comment]
The multipliers are as they should be -- each checklist item is a task on
top of the task itself. The task itself should be counted because for many
folks, that's how it's used -- as a final item or as a final celebration or
culmination. This is the reason that tasks don't auto-check once you've
ticked off all the checklist items.

The unchecking thing is something else entirely -- it's unrelated to the
checklists. The checklists just make it more apparent because of the
multiplier. I can look it up in a bit to give you the ticket number, but
basically with the way things are stored, it's kind of hard for the system
to 100% know what value you gained from the task in the first place, so
unticking it doesn't always remove the correct amount. Multiply that times
10, and it's a lot more apparent.

**Comment by GrayDays** [comment]
Got it, thanks for the answer.

**Comment by deilann** [comment]
Here it is: https://github.com/HabitRPG/habitrpg/issues/2222 :)
